go-controller/pkg/ovn/healthcheck/egressip_healthcheck.go

```diff
@@ -70,13 +73,29 @@ func (ehs *egressIPHealthServer) Run(stopCh <-chan struct{}) {
 	if cfg.Cert == "" || cfg.PrivKey == "" {
 		klog.Warning("Health checking using insecure connection")
 	} else {
-		// Enable TLS for all incoming connections.
-		cert, err := tls.LoadX509KeyPair(cfg.Cert, cfg.PrivKey)
+		// certProvider is responsible for reloading the certificate if it rotates.
+		certProvider, err := pemfile.NewProvider(pemfile.Options{
+			CertFile:        cfg.Cert,
+			KeyFile:         cfg.PrivKey,
+			RefreshDuration: time.Minute,
```

Comment:
Nothing against 1 minute, but what's the reasoning for the 1 hour default not being appropriate?

Reply:
It is a corner case scenario but this is going to help with suspended clusters...
See: https://github.com/ovn-org/ovn-kubernetes/pull/4118 for a similar issue.

Reply:
hmm but that was specific to cert manager and the mechanism it used to wait for rotation. How does that apply here?